File: apps/api/src/main/java/br/unifor/costify/application/contracts/RecipeRepository.java
package br.unifor.costify.application.contracts;

import br.unifor.costify.domain.entity.Recipe;
import br.unifor.costify.domain.valueobject.Id;
import java.util.Optional;

public interface RecipeRepository {

  Recipe save(Recipe recipe);

  Optional<Recipe> findById(Id id);

  boolean existsByName(String name);

  void deleteById(Id id);
}

File: internal/application/contract/recipe_repository.go
package contract

import (
	"github.com/vini/costify-go/internal/domain/entity"
	"github.com/vini/costify-go/internal/domain/valueobject"
)

// RecipeRepository defines the contract for recipe persistence operations
type RecipeRepository interface {
	Save(recipe *entity.Recipe) (*entity.Recipe, error)
	FindById(id valueobject.Id) (*entity.Recipe, error)
	ExistsByName(name string) (bool, error)
	DeleteById(id valueobject.Id) error
}
